sdk/src/invariant.ts

```diff
@@ -4,105 +4,67 @@ import { WeightV2 } from '@polkadot/types/interfaces'
 import { IKeyringPair } from '@polkadot/types/types/interfaces'
 import { DeployedContract } from '@scio-labs/use-inkathon'
 import { deployContract } from '@scio-labs/use-inkathon/helpers'
-import { Network } from './network.js'
 import { InvariantQuery, InvariantTx } from './schema.js'
+import { DEFAULT_PROOF_SIZE, DEFAULT_REF_TIME, sendQuery, sendTx } from './utils.js'
 
 export class Invariant {
-  public static readonly DEFAULT_REF_TIME = 100000000000
-  public static readonly DEFAULT_PROOF_SIZE = 100000000000
-
   contract: ContractPromise | null = null
   api: ApiPromise
-  account: IKeyringPair
-  weight: WeightV2
-  waitForFinalization: boolean
+  gasLimit: WeightV2
+  storageDepositLimit: number | null
 
   constructor(
     api: ApiPromise,
-    account: IKeyringPair,
-    network: Network,
-    refTime: number = Invariant.DEFAULT_REF_TIME,
-    proofSize: number = Invariant.DEFAULT_PROOF_SIZE
+    storageDepositLimit: number | null = null,
+    refTime: number = DEFAULT_REF_TIME,
+    proofSize: number = DEFAULT_PROOF_SIZE
   ) {
     this.api = api
-    this.account = account
-    this.weight = api.registry.createType('WeightV2', {
+    this.gasLimit = api.registry.createType('WeightV2', {
       refTime,
       proofSize
     }) as WeightV2
-    this.waitForFinalization = network != Network.Local
```

Comment:
Overall, `waitForFinalization` should remain as is - my mistake. However, if not provided by the user, it fully depends on the network, as here. Then in send tx:
```Typescript
if (result.isCompleted && !waitForFinalization) {
    resolve(result.txHash.toHex();
}

if (result.isFinalized) {
    resolve(result.txHash.toHex();
}
```